database-commons/src/main/java/io/cdap/plugin/db/CommonSchemaReader.java

```diff
@@ -127,4 +128,23 @@ public Schema getSchema(ResultSetMetaData metadata, int index) throws SQLExcepti
 
     return Schema.of(type);
   }
+
+  @Override
+  public boolean ignoreColumn(ResultSetMetaData metadata, int index) throws SQLException {
+    return false;
+  }
+
+  @Override
+  public boolean isTypeCompatible(Schema.Field field, ResultSetMetaData metadata, int index) throws SQLException {
+    int type = metadata.getColumnType(index);
+    int precision = metadata.getPrecision(index);
+    int scale = metadata.getScale(index);
+
+    Schema.Type fieldType = field.getSchema().isNullable() ? field.getSchema().getNonNullable().getType()
+      : field.getSchema().getType();
+    Schema columnSchema = DBUtils.getSchema(type, precision, scale);
+    Schema.Type columnType = columnSchema.getType();
+
+    return Objects.equals(fieldType, columnType);
+  }
```

Comment:
Field's type validation of the given input schema moved from the `AbstractDBSink` to `CommonSchemaReader`, so now it can be customized.